k8s.gcr.io/ensure-main-project.sh

```diff
@@ -0,0 +1,172 @@
+color 6 "Enabling the OSLogin API"
+enable_api "${PROJECT}" oslogin.googleapis.com
+
+# Make an account for GKE nodes to run as
+color 6 "Creating service account for ${NODES_SVCACCT}"
+ensure_service_account "${PROJECT}" "${NODES_SVCACCT}" "Least-privilege SA for k8s nodes"
+
+color 6 "Empowering ${NODES_SVCACCT} with min permissions"
+acct=$(svc_acct_for "${PROJECT}" "${NODES_SVCACCT}")
+gcloud projects add-iam-policy-binding "${PROJECT}" \
+    --member "serviceAccount:${acct}" \
+    --role roles/logging.logWriter
+gcloud projects add-iam-policy-binding "${PROJECT}" \
+    --member "serviceAccount:${acct}" \
+    --role roles/monitoring.viewer
+gcloud projects add-iam-policy-binding "${PROJECT}" \
+    --member "serviceAccount:${acct}" \
+    --role roles/monitoring.metricWriter
+
+color 6 "Empowering BigQuery admins"
+gcloud projects add-iam-policy-binding "${PROJECT}" \
+    --member "group:${BQ_ADMINS_GROUP}" \
+    --role roles/bigquery.admin
+
+color 6 "Empowering cluster admins"
+gcloud projects add-iam-policy-binding "${PROJECT}" \
+    --member "group:${CLUSTER_ADMINS_GROUP}" \
+    --role roles/compute.viewer
+gcloud projects add-iam-policy-binding "${PROJECT}" \
+    --member "group:${CLUSTER_ADMINS_GROUP}" \
+    --role roles/container.admin
+if ! gcloud --project "${PROJECT}" iam roles describe ServiceAccountLister >/dev/null 2>&1; then
+    # Don't use `yes` here, it causes SIGPIPE -> pipefail -> errexit
+    echo "y" | gcloud --project "${PROJECT}" \
+        iam roles create ServiceAccountLister \
+        --title "Service Account Lister" \
+        --description "Can list ServiceAccounts." \
+        --stage GA \
+        --permissions iam.serviceAccounts.list
+fi
+gcloud projects add-iam-policy-binding "${PROJECT}" \
+    --member "group:${CLUSTER_ADMINS_GROUP}" \
+    --role "projects/${PROJECT}/roles/ServiceAccountLister"
+
+color 6 "Empowering GCP accounting"
+gcloud projects add-iam-policy-binding "${PROJECT}" \
+    --member "group:${ACCOUNTING_GROUP}" \
+    --role roles/bigquery.jobUser
+
+color 6 "Creating the BigQuery dataset for billing data"
+if ! bq --project "${PROJECT}" ls "${BQ_BILLING_DATASET}" >/dev/null 2>&1; then
+    bq --project "${PROJECT}" mk "${BQ_BILLING_DATASET}"
+fi
+
+color 6 "Setting BigQuery permissions"
+
+CUR=$(tempfile -p k8s-infra-bq-access-cur)
+bq show --format=prettyjson "${PROJECT}":"${BQ_BILLING_DATASET}"  > "${CUR}"
+
+ENSURE=$(tempfile -p k8s-infra-bq-access-new)
+cat > "${ENSURE}" << __EOF__
+{
+  "access": [
+    {
+      "groupByEmail": "${ACCOUNTING_GROUP}",
+      "role": "READER"
+    },
+    {
+      "groupByEmail": "${ACCOUNTING_GROUP}",
+      "role": "roles/bigquery.metadataViewer"
+    },
+    {
+      "groupByEmail": "${ACCOUNTING_GROUP}",
+      "role": "roles/bigquery.user"
+    }
+  ]
+}
+__EOF__
+
+FINAL=$(tempfile -p k8s-infra-bq-access-new)
+jq -s '.[0].access + .[1].access | { access: . }' "${CUR}" "${ENSURE}" > "${FINAL}"
```

Comment:
A comment would be great here to explain what is happening!  We're merging the existing and new permissions?  (And meta: any reason not just to set the permissions explicitly so we do delete any unwanted permissions?)

Reply:
comment added

Reply:
@thockin I don't see the comment. forgot to push?

Reply:
Weird, I swear I wrote a comment.  Added now.